File: do_it_python/boj/10828.py
import sys

n = int(sys.stdin.readline())

stack = []

for i in range(n):
    command = sys.stdin.readline().split()
    # push
    if command[0] == 'push':
        stack.append(command[1])
    # pop
    elif command[0] == 'pop':
        # stack.pop() ? print(stack.pop()) : print("-1")
        if len(stack) == 0:
            print(-1)
        else:
            print(stack.pop())
    # size
    elif command[0] == 'size':
        print(len(stack))
    # empty
    elif command[0] == 'empty':
        if len(stack) == 0:
            print("1")
        else:
            print("0")

    # top
    else:
        if len(stack) == 0:
            print("-1")
        else:
            print(stack[-1])

File: c_boj/10828.c
#define _CRT_SECURE_NO_WARNINGS
#include <string.h>
#include <stdio.h>

int stack[100001];
int count = 0;
// !! void -> return이 없는 자료형
//push
void push(int n)
{
  stack[count] = n;
  count++;
}

//pop
void pop()
{
  if (count == 0)
    printf("%d\n", -1);
  else
  {
    count--;
    printf("%d\n", stack[count]);
    stack[count] = 0;
  }
}
//size
void size()
{
  printf("%d\n", count);
}
//empty
void empty()
{
  if (count == 0)
    printf("%d\n", 1);
  else
    printf("%d\n", 0);
}

//top
void top()
{
  if (count == 0)
    printf("%d\n", -1);
  else
    printf("%d\n", stack[count - 1]);
}

int main()
{
  int n;
  char number[10];

  scanf("%d", &n);

  // int stack[100];
  for (int i = 0; i < n; i++)
  {
    scanf("%s", number);
    if (strcmp(number, "push") == 0)
    {
      int num;
      scanf("%d", &num);
      push(num);
    }
    else if (strcmp(number, "pop") == 0)
      pop();
    else if (strcmp(number, "top") == 0)
      top();
    else if (strcmp(number, "size") == 0)
      size();
    else
      empty();
  }
}